### ISSUE TITLE
"Divide by zero" warning when `difference` is applied

### ISSUE THREAD

**Issue by theroggy** [libgeos/geos#1048]
For some (rare) combinations of valid polygons, applying a `difference` between them gives a "divide by zero" warning.

Not a huge issue, as the result seems OK, but...

Script to reproduce:
``` py
from matplotlib import pyplot as plt
import shapely
import shapely.plotting

wkt1 = "POLYGON ((142388.68 186162.28, 142388.68 186262.28, 142405.12908847188 186262.28, 142396.74 186247.45, 142410.48 186240.2, 142413.06999999998 186233.67000000004, 142415.20546765794 186233.96797223136, 142417.50658364198 186228.23343848815, 142431.75 186238.95, 142437.12000000002 186244.67000000004, 142437.16999999998 186244.71999999997, 142447.93 186256.2, 142452.99739099355 186262.28, 142491.8244192841 186262.28, 142493.22999999943 186261.61000000025, 142525.995 186246.7523287307, 142525.995 186184.16700622076, 142522.81999999846 186178.7699999973, 142522.6899999992 186178.5399999986, 142514.03 186163.82, 142513.16484503727 186162.28, 142428.13469668917 186162.28, 142424.62500000003 186164.551, 142422.9949373521 186162.28, 142388.68 186162.28))"
wkt2 = "MULTIPOLYGON (((142388.68 186214.8177165806, 142388.68 186262.28, 142405.12908847185 186262.28, 142396.74 186247.45, 142410.48 186240.2, 142413.07 186233.67, 142413.93 186233.79, 142413.253 186233.302, 142415 186231.92, 142388.68 186214.8177165806)), ((142388.68 186162.28, 142388.68 186207.74701303276, 142391.03 186209.53, 142406.39 186220.65, 142416.04 186227.13, 142431.75 186238.95, 142437.12 186244.67, 142437.17 186244.72, 142447.93 186256.2, 142452.99739099355 186262.28, 142491.82441928412 186262.28, 142493.23 186261.61, 142525.995 186246.75232873068, 142525.995 186184.1670062208, 142522.82 186178.77, 142522.69 186178.54, 142514.03 186163.82, 142513.16484503727 186162.28, 142428.13469668914 186162.28, 142424.625 186164.551, 142422.99493735208 186162.28, 142388.68 186162.28)))"
poly1 = shapely.from_wkt(wkt1)
poly2 = shapely.from_wkt(wkt2)

print(f"{poly1.is_valid=}")
print(f"{poly2.is_valid=}")

diff = shapely.difference(poly1, poly2)

print(f"{shapely.get_num_coordinates(diff)=}")

shapely.plotting.plot_polygon(diff, facecolor="green", edgecolor="green", add_points=False, alpha=0.3)
shapely.plotting.plot_polygon(poly1, facecolor="none", edgecolor="blue", add_points=False, hatch="\\")
shapely.plotting.plot_polygon(poly2, facecolor="none", edgecolor="red", add_points=False, hatch="/")
plt.show()
```

**Comment by pramsey** [comment]
Could be dupe of #515

**Comment by pramsey** [comment]
And I don't see any warnings off this scenario when run in the C++ unit test framework 
```c++
template<>
template<>
void object::test<46> ()
{
    set_test_name("testPolygonDifferenceDivZero");
    std::string a = "POLYGON ((142388.68 186162.28, 142388.68 186262.28, 142405.12908847188 186262.28, 142396.74 186247.45, 142410.48 186240.2, 142413.06999999998 186233.67000000004, 142415.20546765794 186233.96797223136, 142417.50658364198 186228.23343848815, 142431.75 186238.95, 142437.12000000002 186244.67000000004, 142437.16999999998 186244.71999999997, 142447.93 186256.2, 142452.99739099355 186262.28, 142491.8244192841 186262.28, 142493.22999999943 186261.61000000025, 142525.995 186246.7523287307, 142525.995 186184.16700622076, 142522.81999999846 186178.7699999973, 142522.6899999992 186178.5399999986, 142514.03 186163.82, 142513.16484503727 186162.28, 142428.13469668917 186162.28, 142424.62500000003 186164.551, 142422.9949373521 186162.28, 142388.68 186162.28))";
    std::string b = "MULTIPOLYGON (((142388.68 186214.8177165806, 142388.68 186262.28, 142405.12908847185 186262.28, 142396.74 186247.45, 142410.48 186240.2, 142413.07 186233.67, 142413.93 186233.79, 142413.253 186233.302, 142415 186231.92, 142388.68 186214.8177165806)), ((142388.68 186162.28, 142388.68 186207.74701303276, 142391.03 186209.53, 142406.39 186220.65, 142416.04 186227.13, 142431.75 186238.95, 142437.12 186244.67, 142437.17 186244.72, 142447.93 186256.2, 142452.99739099355 186262.28, 142491.82441928412 186262.28, 142493.23 186261.61, 142525.995 186246.75232873068, 142525.995 186184.1670062208, 142522.82 186178.77, 142522.69 186178.54, 142514.03 186163.82, 142513.16484503727 186162.28, 142428.13469668914 186162.28, 142424.625 186164.551, 142422.99493735208 186162.28, 142388.68 186162.28)))";
    std::string exp = "MULTIPOLYGON(((142388.68 186214.8177165806,142415 186231.92,142413.253 186233.302,142413.93 186233.79,142413.07 186233.67,142413.07 186233.67000000004,142415.20546765794 186233.96797223136,142417.50658364198 186228.23343848815,142431.75 186238.95,142416.04 186227.13,142406.39 186220.65,142391.03 186209.53,142388.68 186207.74701303276,142388.68 186214.8177165806)),((142405.12908847188 186262.28,142396.74 186247.45,142405.12908847185 186262.28,142405.12908847188 186262.28)),((142447.93 186256.2,142437.17 186244.72,142437.16999999998 186244.71999999997,142447.93 186256.2)),((142525.995 186246.7523287307,142525.995 186246.75232873068,142495.35154412367 186260.64796108124,142525.995 186246.7523287307)),((142525.995 186184.16700622076,142522.8199999992 186178.76999999856,142522.82 186178.77,142525.995 186184.1670062208,142525.995 186184.16700622076)),((142422.9949373521 186162.28,142422.99493735208 186162.28,142424.625 186164.551,142424.62500000003 186164.551,142422.9949373521 186162.28)))";
    testOverlay(a, b, exp, OverlayNG::DIFFERENCE, 0);
}
```

**Comment by theroggy** [comment]
> Could be dupe of #515

It is definitely related, but based on https://github.com/libgeos/geos/pull/806 #515 was fixed, so if that's correct, it will have to be another code path giving the same warning...

**Comment by pramsey** [comment]
Paging @jorisvandenbossche, as I never found a good way to trace these.

**Comment by jorisvandenbossche** [comment]
I wanted to take a look, but I can't actually reproduce it with shapely built with GEOS main. Not seeing any warning, so that confirms @pramsey's comment above

**Comment by jorisvandenbossche** [comment]
And I _do_ see the warning in my env with released GEOS, which is using GEOS 3.12.1. So this appears to have been "fixed" on main
